server/s3_get_object_action.cc

```diff
@@ -132,6 +132,22 @@ void S3GetObjectAction::fetch_object_info_failed() {
 
 void S3GetObjectAction::validate_object_info() {
   s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  if (object_metadata->is_delete_marker()) {
+    s3_log(S3_LOG_DEBUG, request_id, "Object is a delete marker.");
+    request->set_out_header_value("x-amz-delete-marker", "true");
+    request->set_out_header_value("x-amz-version-id",
+                                  object_metadata->get_obj_version_id());
+
+    if (request->has_query_param_key("versionId")) {
+      request->set_out_header_value("Allow", "DELETE");
```

Comment:
This response header does not seem to be documented anywhere. Is it correctly to just blindly set it? How do we know when it's set to this value? How do you know a DELETE is allowed on the object being returned? Could it be affected by IAM permissions?

Reply:
OK, I found some documentation, I guess it's a standard response header when "MethodNotAllowed" is returned.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Allow

> This header must be sent if the server responds with a 405 Method Not Allowed status code to indicate which request methods can be used.

Still not sure if we're setting it properly, but maybe OK for now?

Reply:
I'd probably document this in the Observations page and or LLD. Also might want to try something like perform a Get with an IAM account where the permission allows a Get, but not a Delete, and see if the header is empty (`Allow:`) instead.

Reply:
Updated the [Observation](https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/817889369/Observations#MethodNotAllowed) document.

Reply:
@keithpine @sjain09 Resolving this comment since it has been answered. Please create a follow-up ticket if needed.